paasta_tools/tron_tools.py

```diff
@@ -650,6 +650,15 @@ def format_master_config(master_config, default_volumes, dockercfg_location):
         }
     )
     master_config["mesos_options"] = mesos_options
+
+    k8s_options = master_config.get("k8s_options", {})
+    if k8s_options:
+        # Only add default volumes if we already have k8s_options
+        # TODO: Gate this on whether k8s on tron is enabled
+        k8s_options.update(
+            {"default_volumes": format_volumes(default_volumes),}
+        )
```

Comment:
How come we also don't add dockercfg_location in this case?

Reply:
afaik, we've configured k8s to know how to talk to our internal registry so we don't need to pass any additional config for that

Reply:
+1 to the above -- The contents of dockercfg_location gets thrown into mesos task's `uris` field, so it basically just controls how we talk to various docker registries it seems, which our kubernetes clusters already know how to do.